### DISCUSSION TITLE
"Store" for templates?

### DISCUSSION THREAD

**Discussion by LiaraAlis** [kimai/kimai#3556] [General]
Hi,

I've made a new invoice template to match the requirements of DIN 5008 as good as possible that looks a little bit more modern as the default templates shipped with kimai. I'm not an expert for this DIN norm but I did my best. 😁

So I'm thinking about sharing this template with others. But what's the best way for this?
I'm thinking about creating a plugin for this template to make some options configurable (at the moment some variables can be overridden by extending the template) or to create a new repository.

Can we add a overview for templates on the current page?
There is a repository https://github.com/Keleo/kimai2-invoice-templates but it seems to very outdated.
What's about creating a new repository for templates so everyone can do PRs to add templates to that repository?

**Comment by kevinpapst** [comment]
I created a new repository here: https://github.com/kimai/invoice-templates
Having a plugin for a single invoice template is possible, but sounds like overhead to me.

I will inspect and then move everything from the [Keleo repository](https://github.com/Keleo/kimai2-invoice-templates) to the one [I just created](https://github.com/kimai/invoice-templates) and then finally archive the other one.

My idea is to create a directory for each template (?) and add a sentence about it to the README.
The directory should contain the template and an optional markdown file to explain further template options.
I'd say: templates should be self-containing. If they have options, explain them with `{# comments #}` and add some docs to the README. Maybe create a second file that demonstrates how to create multiple templates extending each other.

In the main README we could create lists that separate by type (PDF, ODS, XLSX).

That's my quick idea for that topic. What do you think?

**Reply by LiaraAlis** [reply]
That's sounds like what I thought. I think this the best way to do this.
So we should add the new repository to the docs (maybe after some templates are added).

**Reply by kevinpapst** [reply]
Exactly, one step after another. Let's prepare the new repository and README with at least two templates and then update docs and share it at Twitter. I am just awfully busy this week, both with private and business tasks, so I will need a couple of days.

**Reply by LiaraAlis** [reply]
Fine! :)
I created a PR draft https://github.com/kimai/invoice-templates/pull/1 so we can determine with this the standard for this repository. 😊
So I already wrote something about the template. Maybe this is already going in the right direction.

**Reply by kevinpapst** [reply]
I moved all available temapltes, and shared the link to the repository via twitter.
Thanks for sharing your template @LiaraAlis - I hope more people follow your example 😄

**Reply by LiaraAlis** [reply]
@kevinpapst You still need to adjust the link to the template directory in the documentation: https://www.kimai.org/documentation/invoices.html#twig-templates
It still refers to the old repository. :)

**Reply by kevinpapst** [reply]
True, updated (might take some minutes to show up).
